sunpy/coordinates/tests/test_frames.py

```diff
@@ -176,13 +176,13 @@ def test_hpc_distance_off_limb():
     assert hpc1.Tx == 1500 * u.arcsec
     assert hpc1.Ty == 0 * u.arcsec
 
-    with pytest.raises(SunpyUserWarning, match="is all NaNs"):
+    with pytest.warns(SunpyUserWarning, match="is all NaNs"):
```

Comment:
This happens *all the time* is there a way we can make this error on our regular test suite?

Reply:
We tell pytest to promote warnings to errors so technically a `SunpyUserWarning` (a subclass of `Exception`) is *raised*. We could create a wrapper for (or monkeypatch?) `pytest.raises` such that it always raises an error if the class is a subclass of `Warning`, although that would require people to actually use the pytest wrapper.

Reply:
Yeah, that's the issue. I guess the easy way would be to make one of our test jobs not throw warnings as errors?